lib/rex/post/meterpreter/extensions/stdapi/sys/registry.rb

```diff
@@ -195,10 +212,55 @@ def Registry.set_value(hkey, name, type, data)
     return true
   end
 
+  def Registry.set_value_direct(root_key, base_key, name, type, data, perm = KEY_WRITE)
+    request = Packet.create_request('stdapi_registry_set_value_direct')
+
+    request.add_tlv(TLV_TYPE_ROOT_KEY, root_key)
+    request.add_tlv(TLV_TYPE_BASE_KEY, base_key)
+    request.add_tlv(TLV_TYPE_PERMISSION, perm)
+    request.add_tlv(TLV_TYPE_VALUE_NAME, name)
+    request.add_tlv(TLV_TYPE_VALUE_TYPE, type)
+
+    if (type == REG_SZ)
+      data += "\x00"
+    elsif (type == REG_DWORD)
+      data = [ data.to_i ].pack("V")
+    end
+
+    request.add_tlv(TLV_TYPE_VALUE_DATA, data)
+
+    response = client.send_request(request)
+
+    return true
+  end
+
   #
   # Queries the registry value supplied in name and returns an
   # initialized RegistryValue instance if a match is found.
   #
+  def Registry.query_value_direct(root_key, base_key, name, perm = KEY_READ)
+    request = Packet.create_request('stdapi_registry_query_value_direct')
+
+    request.add_tlv(TLV_TYPE_ROOT_KEY, root_key)
+    request.add_tlv(TLV_TYPE_BASE_KEY, base_key)
+    request.add_tlv(TLV_TYPE_PERMISSION, perm)
+    request.add_tlv(TLV_TYPE_VALUE_NAME, name)
+
+    response = client.send_request(request)
+
+    type = response.get_tlv(TLV_TYPE_VALUE_TYPE).value
+    data = response.get_tlv(TLV_TYPE_VALUE_DATA).value
+
+    if (type == REG_SZ)
+      data = data[0..-2]
+    elsif (type == REG_DWORD)
```

Comment:
You can omit parentheses around the condition https://github.com/bbatsov/ruby-style-guide#no-parens-if